File: typescript/easy/reverse-vowels-of-a-string.ts
function reverseVowels(s: string): string {
    let chars = s.split("");
    let n = chars.length;
    const vowels = new Set("aeiouAEIOU");

    for (let i = 0; i < n; i++) {
        console.log(n);
        if (vowels.has(chars[i])) {
            for (let j = n - 1; j > i; j--) {
                n -= 1;
                if (vowels.has(chars[j])) {
                    [chars[i], chars[j]] = [chars[j], chars[i]];
                    break;
                }
            }
        }
    }

    return chars.join("");
}

File: csharp/easy/reverse-vowels-of-a-string.cs
string ReverseVowels(string s)
{
    char[] chars = s.ToCharArray();
    int n = chars.Length;
    HashSet<char> vowels = new HashSet<char>("aeiouAEIOU".ToCharArray());

    for (int i = 0; i < n; i++)
    {
        Console.WriteLine(n);
        if (vowels.Contains(chars[i]))
        {
            for (int j = n - 1; j > i; j--)
            {
                n--;
                if (vowels.Contains(chars[j]))
                {
                    char temp = chars[i];
                    chars[i] = chars[j];
                    chars[j] = temp;
                    break;
                }
            }
        }
    }

    return new string(chars);
}
